sogs/templates/base.html

```diff
@@ -1,9 +1,79 @@
-<html>
+<html id="html" class="dark">
   <head>
+    
     <title> sogs </title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="{{ url_for('static', filename='output.css') }}">
   </head>
-  <body>
+  <body class="dark:bg-otherGray transition duration-500 h-full flex flex-col overflow-hidden">
+    
+    {% block modal %}
+    {% endblock %}
+    
+    <!-- header / navbar -->
+    <div class="px-2 sm:px-5 flex-none flex justify-around sm:justify-between items-center w-full h-32 max-h-32 
+    bg-white dark:bg-otherGray border-b-2 border-black dark:border-lightGray transition duration-500">
+      <div class="text-black text-center">
+        <button>
+          <a href="/">
+            <h1 class="  dark:text-white text-3xl xs:text-4xl sm:text-5xl decoration-black dark:decoration-white decoration-2">{% block heading %}{% endblock %}</h1>
+            <div class="flex flex-col sm:flex-row pl-3 sm:pl-0">
+              <p class=" font-bold text-sessionGreen dark:text-sessionGreen text-lg xs:text-xl sm:text-2xl">Session Open&nbsp</p>
+              <p class=" font-bold text-sessionGreen dark:text-sessionGreen text-lg xs:text-xl sm:text-2xl">Group Server</p>
+            </div>
+          </a>
+        </button>
+      </div>
+      
+      <div class="flex space-x-4 ">
+        {% block invite %}
+        {% endblock %}
+        <button id="moonIcon" class='px-1 rounded' onclick="toggleDarkMode()"><img src="{{ url_for('static', filename='moon-d-50-min.png') }}" alt="moon"></button>
+        <button id="sunIcon" style="display: none;" class='px-1 rounded' onclick="toggleDarkMode()"><img src="{{ url_for('static', filename='sun-w-50-min.png') }}" alt="sun"></button>
+      </div>
+    </div>
+    
+    <!-- content/body -->
     {% block body %}
     {% endblock %}
   </body>
+  <!-- dark mode toggle js -->
+  <script>
+      var htmlElement = document.getElementById("html");
+      var moonIcon = document.getElementById("moonIcon");
+      var sunIcon = document.getElementById("sunIcon");
+
+      function setPageTheme(){
+        if (localStorage.theme === 'dark' || (!('theme' in localStorage) && window.matchMedia('(prefers-color-scheme: dark)').matches)) {
+          htmlElement.classList.add('dark')
+          sunIcon.style.display = 'block'
+          moonIcon.style.display = 'none'
+        } else {
+          htmlElement.classList.remove('dark')
+          sunIcon.style.display = 'none'
+          moonIcon.style.display = 'block'
+        }
+      }
+      
+      function toggleDarkMode(){
+        if ('theme' in localStorage && localStorage.theme === 'light') {
+          localStorage.theme = 'dark'
+          setPageTheme();
+        } else {
+          localStorage.theme = 'light'
+          setPageTheme();
+        }
+      }
```

Comment:
default theme is light

first click switch to light again
second click switch to dark

you can test this in an incognito window
easy fix would be I think
```
        if (!localStorage.theme || localStorage.theme === 'light') {
```

Reply:
yeah i was unsure on this - did it this way as a quick fix to avoid flash of light mode when set to dark theme - i thought its better to default to dark than to have light mode flash every time for a dark mode user - ill look into it some more and change